runtime/src/main/java/io/quarkiverse/jasperreports/repository/DefaultReadOnlyRepository.java

```diff
@@ -0,0 +1,115 @@
+package io.quarkiverse.jasperreports.repository;
+
+import static io.quarkiverse.jasperreports.Constants.EXT_COMPILED;
+import static io.quarkiverse.jasperreports.Constants.EXT_DATA_ADAPTER;
+import static io.quarkiverse.jasperreports.Constants.EXT_STYLE;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Optional;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.Dependent;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+
+import io.quarkiverse.jasperreports.Constants;
+import net.sf.jasperreports.engine.JRException;
+import net.sf.jasperreports.engine.JasperReportsContext;
+import net.sf.jasperreports.engine.SimpleJasperReportsContext;
+import net.sf.jasperreports.engine.util.JRLoader;
+import net.sf.jasperreports.repo.FileRepositoryPersistenceServiceFactory;
+import net.sf.jasperreports.repo.FileRepositoryService;
+import net.sf.jasperreports.repo.PersistenceService;
+import net.sf.jasperreports.repo.PersistenceServiceFactory;
+import net.sf.jasperreports.repo.PersistenceUtil;
+import net.sf.jasperreports.repo.RepositoryService;
+import net.sf.jasperreports.repo.Resource;
+import net.sf.jasperreports.repo.SimpleRepositoryContext;
+import net.sf.jasperreports.repo.StreamRepositoryService;
+
+@Dependent
```

Comment:
Should this be ApplicationScoped?

Reply:
I was thinking that we might leak things from one report invocation to the next, so explicitly making it `@Dependent` would help isolate it.  I could be totally misunderstanding things too, and if we find that nothing is shared between invocations then `@ApplicationScoped` seems correct in that case.